### DISCUSSION TITLE
Budibase start fail due to x509 error

### DISCUSSION THREAD

**Discussion by itworksmart** [Budibase/budibase#3232] [Help] [answered]
I try to self-host budibase on my NAS linux server.

Follow Budibase Cli setup tutorial:
https://docs.budibase.com/self-hosting/self-host/budibase-cli-setup

I get x509 error when last step `budi hosting --start`

I've check system time is updated.

Is there anyone can give a hint on how to solve this?

Thanks.

![budibase start fail](https://user-images.githubusercontent.com/62452539/140014704-1a211307-7d1c-41fc-b9fa-7f93bd815600.PNG)

**Comment by mjashanks** [comment]
That's a strange one. I just checked the certificate that it talks about, and it seems to be ok. It's maybe worth giving it another go - we've seen a few issues with `scarf.sh` recently, which is where the docker images are pulled from. I also just pulled the images successfully.

Try again and let us know!

**Reply by itworksmart** [reply]
I try to install budibase in my synology nas.
Just try a gain and shows same error.

I would like to know if I need to add permission or add some link to tusted resources or not.
I'm newbie to use docker.

I'll try budibase cli on a different VM later.

**Reply by itworksmart** [reply]
Install by ssh login to my device.

All instructions in budibase cli before start is successfully as tutorial.

**Reply by mjashanks** [reply]
Yea, I'm not 100% if Budibase will run well on NAS. I know some people have done it before, but I don't now what they had to do to make it work!

If you want a quick way to try Budibase, you should signup to our cloud platform https://budibase.com - then you could move everything to your VM later, if you like it!

**Comment by itworksmart** [comment, marked as answer]
As a long path search for possible solution. I find a post:
https://issueexplorer.com/issue/Budibase/budibase/2759

https://github.com/Budibase/budibase/issues/2759#issuecomment-938043128

Someone suggest to modify docker image source as below:

> ` try updating the image fields in docker-compose.yaml to be:
> 
> budibase/apps
> budibase/worker
> 
> rather than the scarf ones?`

I follow this instructions and finally pass at beginning.

Later I face another error

> `ERROR: for bbworker  Cannot start service worker-service: Bind mount failed: '/volume1/homes/thomaschen/budibase/logs' does not exists
> 
> ERROR: for worker-service  Cannot start service worker-service: Bind mount failed: '/volume1/homes/thomaschen/budibase/logs' does not exists`

And I manually add a "logs" folder(directory) in my budibase location.

Then I can start successfully and able to login http://localhost:10000

I don't know why but it is okay for me to start.

1. Change docker image source in `docker-compose.yaml `.
2. Add a `logs` directory manually.


Maybe need to modify Budibase Cli script.

**Reply by mjashanks** [reply]
Thanks so much for writing this up - this will likely help others in future. We'll look into the `logs` issue. As for editing the `docker-compose.yaml` - it's a strange one, because it only seems to be a problem for some people. Maybe `scarf.sh` is getting blocked by your company firewall?

Thank you!
